integration-tests/src/query_params.test.js

```diff
@@ -93,6 +93,93 @@ describe("for a GET endpoint with no parameters", () => {
   });
 });
 
+const testOptionalEndpoint: Recouple.Endpoint<
+  {
+    x: ?string,
+    y: string
+  },
+  {
+    x: ?string,
+    y: string
+  }
+> = Recouple.endpoint()
+  .fragment("foo")
+  .queryParams({
+    x: T.maybeString,
+    y: T.string
+  });
+
+const testOptionalHandler = async input => {
+  return {
+    x: input.x || null,
+    y: input.y
+  };
+};
+
+describe("for a GET endpoint with optional query parameters", () => {
+  let server;
+  beforeEach(() => {
+    server = TestUtils.makeServer({
+      endpoint: testOptionalEndpoint,
+      handler: testOptionalHandler
+    });
+  });
+  describe("for the recouple client", () => {
+    it("can be called with a query parameter", async () => {
+      const baseURL = `http://localhost:${server.address().port}`;
+      const input = { x: "X", y: "Y" };
+      await RecoupleFetch.safeGet(baseURL, testOptionalEndpoint, input);
+      const expectedURL = `${baseURL}/foo?x=X&y=Y`;
+      expect(fetch).toHaveBeenLastCalledWith(expectedURL);
+    });
+
+    describe("for null parameters", () => {
+      it("will serialize null parameters to the empty string", async () => {
+        const baseURL = `http://localhost:${server.address().port}`;
+        const input = { x: null, y: "Y" };
+        await RecoupleFetch.safeGet(baseURL, testOptionalEndpoint, input);
+        const expectedURL = `${baseURL}/foo?x=&y=Y`;
+        expect(fetch).toHaveBeenLastCalledWith(expectedURL);
+      });
+
+      it("will remove undefined parameters", async () => {
+        const baseURL = `http://localhost:${server.address().port}`;
+        const input = { x: undefined, y: "Y" };
+        await RecoupleFetch.safeGet(baseURL, testOptionalEndpoint, input);
+        const expectedURL = `${baseURL}/foo?y=Y`;
+        expect(fetch).toHaveBeenLastCalledWith(expectedURL);
+      });
+    });
+  });
+
+  describe("for the recouple server", () => {
+    test("server can parse the input when present", async () => {
+      const resp = await fetch(
+        `http://localhost:${server.address().port}/foo?x=X&y=Y`
+      );
+      const json = await resp.json();
+      expect(json).toEqual({ x: "X", y: "Y" });
+    });
+
+    test("server will parse null inputs", async () => {
+      const resp = await fetch(
+        `http://localhost:${server.address().port}/foo?x=&y=Y`
+      );
+      const json = await resp.json();
+      expect(json).toEqual({ x: null, y: "Y" });
```

Comment:
Right now, we're looking at response as returned by the server (as processed by the given handler). Instead, we should be testing what input the server handler sees as transparently as possible.

Do this by making the handler a spy:
```js
const handler = jest.fn(input => {
  ...
});
```
and then asserting `expect(handler).toHaveLastBeenCalledWith(...)`